src/amo/sagas/reviews.js

```diff
@@ -235,9 +253,23 @@ function* manageAddonReview(
     );
 
     yield put(setReview(reviewFromResponse));
+
+    if (savingRating) {
+      yield put(flashReviewMessage(SAVED_RATING));
+    }
+    if (savingReview) {
+      yield put(flashReviewMessage(SAVED_REVIEW));
+    }
+
+    // Make the message disappear after some time.
+    yield new Promise((resolve) => setTimeout(resolve, FLASH_MESSAGE_DURATION));
+    yield put(flashReviewMessage(undefined));
   } catch (error) {
-    log.warn(`Failed to submit review with action ${action.type}: ${error}`);
+    log.warn(
+      `Failed to create/update review with action ${action.type}: ${error}`,
+    );
     yield put(errorHandler.createErrorAction(error));
+    yield put(flashReviewMessage(ABORTED));
```

Comment:
on error, after aborting, should we reset and set back to undefined? or is that overkill?

Reply:
I considered it but with that approach the caller would never know if the saga had aborted or if the message had just faded away. Nothing is relying on the abort message yet so it doesn't matter. The message just needs to change into *something* on error to make the other one fade away.